e2e/bash/test_bash_shell_integration.py

```diff
@@ -60,7 +60,7 @@ def test_shell_integration(complgen_binary_path: Path):
         input = r'''COMP_WORDS=(mycargo +); COMP_CWORD=1; _complgen_jit_mycargo; printf '%s\n' "${COMPREPLY[@]}"'''
         assert get_sorted_bash_completions(usage_file_path, input) == sorted(['+foo', '+bar'])
 
-
+# This test also needs to be tested manually in an interactive shell
```

Comment:
This looks like an ideal candidate for replicating ["pty" (i.e. pseudoterminal) tests](https://github.com/adaszko/complgen/blob/124bc1721d3547386cb1c8dda24e63dc6ba85153/e2e/zsh/test_zsh_pty.py) from zsh.

Reply:
I will leave that to you :stuck_out_tongue:. The zsh pty tests get stuck forever on my machine; I need to replace  `pty_read.read()` with `pty_read.read(1300)` and remove `os.waitpid(pid, 0)` for the tests to pass.

Reply:
Are you running it on Linux?  Pty handling code is quite OS-specific.  It works for me on macOS

Reply:
Yes, Linux.